### ISSUE TITLE
Projection of XamlUICommand.CanExecuteChanged (system XAML) loses generic type

### ISSUE THREAD

**Issue by asklar** [microsoft/CsWinRT#597]
When projecting [Windows.UI.Xaml.Input.XamlUICommand](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.xamluicommand), the [CanExecuteChanged](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.xamluicommand.canexecutechanged) event is projected as `EventHandler` instead of `EventHandler<T>` causing a compile break ("`XamlUICommand` cannot implement `ICommand` because the type of `CanExecuteChanged` is different" or something like that)

Using the latest cswinrt nuget: 1.1.0-prerelease.201111.6

**Comment by asklar** [comment]
CC @manodasanW @kennykerr is this known / could you take a look?

**Comment by nesherhh** [comment]
Same problem with cswinrt nuget: 1.1.4.
This one is related: #112

**Comment by manodasanW** [comment]
@nesherhh Can you expand on the scenario that you are hitting this in?  Are you trying to use inbox XAML rather than WinUI similar to the original filer?

**Comment by nesherhh** [comment]
Yes. Inbox Windows.UI.Xaml.Input.XamlUICommand.CanExecuteChanged

**Comment by angelazhangmsft** [comment]
@nesherhh currently building inbox XAML projections is not prioritized. What scenario are you trying to use this in - UWP?

**Comment by asklar** [comment]
FYI this would be an important scenario for React Native for Windows, we would love to be able to adopt CsWinRT.

**Comment by nesherhh** [comment]
I was trying to build windows community toolkit for .net 5. It almost worked 🙂

**Comment by azchohfi** [comment]
@nesherhh the WCT is already built for .Net5 (WinUI Desktop):
https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/winui
https://www.nuget.org/packages?q=CommunityToolkit.WinUI

**Comment by nesherhh** [comment]
I need it for WPF and Xaml Islands.

**Comment by nesherhh** [comment]
The Toolkit compiles, but the sample not. I'm getting this errors:
`
9>C:\tmp\Microsoft.Toolkit.Win32\Microsoft.Toolkit.Sample.Wpf.App\Program.cs(89,24,89,80): error CS0012: The type 'XamlApplication' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.Toolkit.Win32.UI.XamlHost, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null, ContentType=WindowsRuntime'.
9>C:\tmp\Microsoft.Toolkit.Win32\Microsoft.Toolkit.Sample.Wpf.App\Program.cs(89,24,89,80): error CS0012: The type 'IXamlMetadataProvider' is defined in an assembly that is not referenced. You must add a reference to assembly 'Windows.Foundation.UniversalApiContract, Version=10.0.0.0, Culture=neutral, PublicKeyToken=null, ContentType=WindowsRuntime'.
`

**Comment by azchohfi** [comment]
Xaml Islands and MapControl are both still not supported on WinUI3: https://docs.microsoft.com/en-us/windows/apps/winui/winui3/#missing-platform-features

**Comment by nesherhh** [comment]
I know, that is why I'm not trying winui3, but what is available in windows toolkit.

**Comment by azchohfi** [comment]
The WCT ships as two separate and distinct versions. One based on the OS Xaml (nuget packages named **Microsoft.Toolkit.\***), and a newer one based on WinUI3 (nuget packages named **CommunityToolkit.WinUI.\***).
If you want to use the WCT on WPF, you will need to use the Microsoft.Toolkit.* version, which is based on the OS Xaml, which supports Xaml Islands and still has the MapControls.
You can't use, as of today, the WCT from the CommunityToolkit.WinUI.* in a Xaml Island. This new version is the WCT is built on top of WinUI3, so if WinUI3 doesn't support Xaml Islands (and as of today, it doesn't), you will **not** be able to run the new version of the WCT (CommunityToolkit.WinUI) inside a Xaml Island.

**Comment by nesherhh** [comment]
I know. I want to use OS Xaml (Microsoft.Toolkit.*) with .NET 5.
.NET 5 compiler doesn’t have winmd support anymore. But there is CsWinRT for .NET 5. So maybe it possible to combine CsWinRT and Microsoft.Toolkit.*

**Comment by azchohfi** [comment]
I don't think that is supported, and I really don't think you will be able to make it work with any simple workaround. The runtime itself has changed, so many things that are being referenced will not load properly on net5.
See the note here: https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/xaml-islands#wpf-and-windows-forms-applications
@angelazhangmsft or @marb2000 will know for a fact.

**Comment by nesherhh** [comment]
According to this https://docs.microsoft.com/en-us/dotnet/core/compatibility/interop/5.0/built-in-support-for-winrt-removed only wind part is moved to cswinrt.

**Comment by azchohfi** [comment]
CSWinRT projects WinRT types to work on net5, yes, but the WCT is not a WinRT library. It is a managed C# library, and it doesn't expose WinMDs for most of its packages (I believe the Notifications package is the only one that does). Also, the WCT (the Microsoft.Toolkit.* version) is based on OS Xaml (WUX), which I don't believe will activate properly on net5.

**Comment by asklar** [comment]
System XAML will activate fine, there are just bugs like this one, as well as the hard-coded mapping of certain .net types to WinUI 3 types (like IPropertyChanged*). 

In our scenario we are not ever using any bindings so we don't need the mappings, we just need the code to compile, so we would like CsWinRT to fix this simple XamlUICommand projection bug

**Comment by nesherhh** [comment]
@asklar did you found a workaround or fix?

**Comment by asklar** [comment]
the only workaround is to produce the projection, and edit it by hand

**Comment by nesherhh** [comment]
Thanks, this worked. Now I'm able to use Xaml Islands on .NET 5.
WindowsXamlHost and MapControl are running in the Microsoft.Toolkit.Sample.Wpf.App project.
Because of #39 I had to comment bind for style property.
Only issue I have now is UWP App (with XamlApplication) have to be c++/winrt project. As soon as I access any UWP class in C# UWP project I get PlatformNotSupported exception. Any other custom class in UWP project can be used without problem. Would be great to know what throws platform not supported exception.

**Comment by nesherhh** [comment]
I just leave it here https://github.com/nesherhh/XamlIslandsNET5

**Comment by angelazhangmsft** [comment]
Duplicate of #723

**Comment by mcooley** [comment]
A fix for this issue would be useful to me, even independently of the full system XAML scenario that #723 is tracking.

Here's my scenario: I'm building a command-line tool that uses [Win2D](https://github.com/microsoft/win2D). My tool doesn't use XAML, but the version of Win2D I want to use depends on System XAML. I'd like to use cswinrt to generate bindings for Win2D, but I can't because it pulls in XAML types that hit this bug. The XAML bindings don't actually need to _work_ in my scenario, but it would be super helpful if they at least compiled.

**Comment by asklar** [comment]
@angelazhangmsft, @mcooley has coded up a low risk fix that will unblock the scenario. It should have no impact on anything winappsdk related, so I would like for us to take it.
